avm/res/maintenance/configuration-assignment/main.bicep

```diff
@@ -0,0 +1,109 @@
+metadata name = 'Maintenance Configuration Assignments'
+metadata description = 'This module deploys a Maintenance Configuration Assignment.'
+
+// ============== //
+//   Parameters   //
+// ============== //
+
+@description('Required. Maintenance configuration assignment Name.')
+param name string
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Required. The maintenance configuration resource ID.')
+param maintenanceConfigurationResourceId string
+
+@description('Conditional. The unique virtual machine resource ID to assign the configuration to. Required if filter is not provided. If resourceId is provided, filter is ignored. If provided, the module scope must be set to the resource group of the virtual machine.')
+param resourceId string?
```

Comment:
Would you think it makes sense to name this `virtualMachineResourceId` as per the description - IF - it is always a VM?